packages/yarnpkg-core/sources/Project.ts

```diff
@@ -942,7 +942,8 @@ export class Project {
         const location = ppath.join(fetchResult.packageFs.getRealPath(), fetchResult.prefixPath);
         packageLocations.set(pkg.locatorHash, location);
 
-        if (buildScripts.length > 0) {
+        // Virtual workspaces shouldn't be built as they don't really exist
+        if (!structUtils.isVirtualLocator(pkg) && buildScripts.length > 0) {
```

Comment:
That's only true for workspaces - in the case of regular dependencies, they're unplugged on disk in their own directories.

Reply:
Yes, that's why it's only applied to the workspaces section
https://github.com/yarnpkg/berry/blob/68e37779e72caf8fe1c5580a4e9a659beff1808f/packages/yarnpkg-core/sources/Project.ts#L919-L952